pkg/controllers/provisioning/launcher.go

```diff
@@ -111,6 +122,14 @@ func (l *Launcher) bind(ctx context.Context, node *v1.Node, pods []*v1.Pod) (err
 	return nil
 }
 
+func (l *Launcher) updateState(ctx context.Context, provisioner *v1alpha5.Provisioner) (*v1alpha5.Provisioner, error) {
```

Comment:
I think that this might be dangerous. It's possible that the provisioners configuration has changed in a way that would violate current launch parameters. We're currently very careful about refreshing provisioner config, using a threadsafe cache and ensuring that pods are properly drained on the old configuration.

It looks like the provisioner in-memory isn't updated w/ the latest status because we're shortcircuitting on spec differences in the provisioners controller. Perhaps we could do something like update the status of the provisioner, but I worry a bit about how threadsafe that might be. 

I haven't thought too deeply about it, but we might need to pivot to using an in memory store. If we go this approach, we could update the store after launch and check it before launch. I know we've thrashed a bit on this -- happy to discuss offline.

Reply:
Discussed this offline and changed this logic. We didn't really need to re-assign the provisioner, but rather only fetch the latest provisioner.Status so doing that instead. This should be fine since provisioner.Status is updated fairly quick (off a node watch)